code/modules/antagonists/heretic/heretic_antag.dm

```diff
@@ -393,6 +406,61 @@
 		var/datum/heretic_knowledge/living_heart/heart_knowledge = get_knowledge(/datum/heretic_knowledge/living_heart)
 		heart_knowledge.on_research(source, src)
 
+/// Signal proc for [COMSIG_LIVING_CULT_SACRIFICED] to reward cultists for sacrificing a heretic
+/datum/antagonist/heretic/proc/on_cult_sacrificed(mob/living/source, list/invokers)
+	SIGNAL_HANDLER
+
+	// Create the blade, give it the heretic and a randomly-chosen master for the soul sword component
+	var/obj/item/melee/cultblade/haunted/haunted_blade = new(get_turf(source), source, pick(invokers))
+
+	ASYNC
+		haunted_blade.gender_reveal(outline_color = COLOR_CULT_RED)
+
+	for(var/mob/living/culto as anything in invokers)
+		to_chat(culto, span_cult_large("\"A follower of the forgotten gods! You must be rewarded for such a valuable sacrifice.\""))
+
+	// Locate a cultist team (Is there a better way??)
+	var/mob/living/random_cultist = pick(invokers)
+	var/datum/antagonist/cult/antag = random_cultist.mind.has_antag_datum(/datum/antagonist/cult, TRUE)
+	if(!antag)
+		CRASH("offer invoker has no mind or cult datum")
+	var/datum/team/cult/cult_team = antag.get_team()
+
+	// Unlock one of 3 special items!
+	var/list/possible_unlocks
+	for(var/i in cult_team.unlocked_heretic_items)
+		if(cult_team.unlocked_heretic_items[i] == TRUE)
+			continue
+		LAZYADD(possible_unlocks, i)
+	if(length(possible_unlocks))
+		var/result = pick(possible_unlocks)
+		cult_team.unlocked_heretic_items[result] = TRUE
+
+		for(var/datum/mind/mind as anything in cult_team.members)
+			if(mind.current)
+				SEND_SOUND(mind.current, 'sound/magic/clockwork/narsie_attack.ogg')
+				to_chat(mind.current, span_cult_large(span_warning("Arcane and forbidden knowledge floods your forges and archives. The cult has learned how to create the ")) + span_cult_large(span_hypnophrase("[result]!")))
+
+	return SILENCE_SACRIFICE_MESSAGE|DUST_SACRIFICE
+
+/obj/item/proc/gender_reveal(outline_color = pick(COLOR_ADMIN_PINK, COLOR_BLUE_LIGHT), anim_time = 10 SECONDS)
```

Comment:
Aint no way you have an item level proc titled `gender_reveal` with no documentation